### PR TITLE
Check for error before writing new file in client.Store()

### PR DESCRIPTION
**Pull request by chetanyakan**
**Summary**
Prevent creation of a corrupted file if the file conversion fails.

Explain the **motivation** for making this change. What existing problem does the pull request solve?

I tried converting a ppt file to pdf. Due to some memory or time issue (I wasn't able to figure out the cause), the file failed to convert and a corrupted file of ~57B was written instead, which could not be opened in a PDF viewer.

For reference, the following error was in the Gotenberg logs:
```
code=500, message=unoconv: non-zero exit code: exit status 6
```

I expected the client.Store() method to return an error instead of writing a corrupted file.

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output.

**Checklist**

- [x] Have you followed the guidelines in our [CONTRIBUTING](CONTRIBUTING.md) guide?
- [x] Have you lint your code locally prior to submission (`make lint`)?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally (`make tests`)?
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

**Comment by gulien** [comment]
Hello @chetanyakan 

Thank you for this PR. 

> I expected the client.Store() method to return an error instead of writing a corrupted file.

Indeed, I was expected the line `resp, err := c.Post(req)` to return an error if status != 200 (I'm not a familiar with Golang HTTP client..).

Concerning your conversion issue, did you use the latest version of Gotenberg?

File: client.go [hunk 1]
```diff
@@ -94,6 +94,11 @@ func (c *Client) Store(req Request, dest string) error {
 		return err
 	}
 	defer resp.Body.Close()
+
+  // Check for 2XX Status Codes
+	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
```
Comment (gulien) [inline client.go]: Why not simply check `res.StatutsCode != http.StatusOK`?

Comment (chetanyakan) [reply client.go]: Indeed I was not aware whether `gotenberg` returns any non-200 (2XX) status codes upon successful conversion to pdf.
If the status code is expected to be 200 always, you can let me know and I can make the required change.

**Comment by chetanyakan** [comment]
Regarding my issue, I was using `gotenberg` v4.
It was likely some memory-related issue as although I was able to repetedly see the issue before, I was able to successfully convert the same file afterward with same version of `gotenberg`.

**Comment by gulien** [comment]
> It was likely some memory-related

Version 6 should behave way better on that issue 👍

File: client.go (hunk 1 above)
Comment (gulien) [reply client.go]: Yes the expected status code is always excepted to be 200 😄

Comment (chetanyakan) [reply client.go]: Done.

**Comment by gulien** [comment]
Thank you 😄 👍

**Pull request merged**